fix bug 1436073 - Introduce Photon

andymikulski opened the pull request:
- [Bug #1436073](https://bugzilla.mozilla.org/show_bug.cgi?id=1436073)
- Introduces Photon typography, colors, and general styles
- Includes styling for `select`, `button` elements


---

<img height="400" src="https://user-images.githubusercontent.com/2767162/38747315-b6913854-3f07-11e8-82f5-4f0025403134.png">
<img height="400" src="https://user-images.githubusercontent.com/2767162/38747314-b67d50d2-3f07-11e8-9bc3-e6695861e5a2.png">
<img height="400" src="https://user-images.githubusercontent.com/2767162/38747313-b6676df8-3f07-11e8-939c-2160cfa7f5d0.png">

andymikulski commented on webapp-django/crashstats/crashstats/static/crashstats/css/base/variables.less:
```diff
@@ -1,39 +1,43 @@
 @image-path: "/static/img";
 
-@base-font: Arial, Helvetica, sans-serif;
+@import "https://code.cdn.mozilla.net/fonts/fira.css";
```
I think this is because `variables.less` are imported both in the `base` styles as well as the subpages. The base styles load the font, then the subpage's bundle also loads the font. I've moved this import to `fonts.less` which is imported only in the `base.less` file.

andymikulski commented on webapp-django/crashstats/crashstats/static/crashstats/css/base/tables.less:
```diff
@@ -21,40 +21,45 @@ table {
     /* standard table styles for data tables */
     &.data-table {
```
I've refactored the table stuff a bit, but essentially the only difference with the `data-` and `zebra` tables is they simply get the every-other shading and hover effects. This was part of the issue with the crashes-per-day table, but as I said, should be straightened out now.

andymikulski commented on webapp-django/crashstats/crashstats/static/crashstats/css/base/layout.less:
```diff
@@ -35,59 +35,53 @@ body {
-            height: 36px;
-            background: url("@{image-path}/2.0/title.png") no-repeat top left;
+    a {
+        display: flex;
```
The combination of `display: flex` with `align-items: center;` is the fix - essentially we're just vertically centering the `span`.

andymikulski commented on webapp-django/crashstats/crashstats/static/crashstats/css/base/layout.less:
```diff
@@ -35,59 +35,53 @@ body {
+        display: block;
+    }
+    .icon {
+        display: none; // TODO: display when logo it's been updated for photon.
```
Only concern here is in the (unlikely) event the icon is part of the E2E tests, though I've removed the element/styling all together. Just something to keep in mind, but probably not important.

Osmose commented on webapp-django/crashstats/crashstats/static/crashstats/css/base/tables.less:
```diff
@@ -32,15 +45,8 @@ table {
-            }
-            &:hover {
-                background-color: @light-grey;
-            }
```
<img width="476" alt="screen shot 2018-04-19 at 1 35 03 pm" src="https://user-images.githubusercontent.com/193106/39016887-94fa18cc-43d6-11e8-93ff-019e287f6c9c.png">

awwwwwww yiss

andymikulski commented:
>It actually looks decent here as well, was this intentional?

<img src="https://user-images.githubusercontent.com/2767162/39153730-38a61a40-4709-11e8-9f0e-dabc613ee602.gif" height=200 />

---

I've fixed the button issue on the `Reports` page - there was some outdated CSS breaking the padding. Also, the header line height has been adjusted a bit (so wrapped text isn't so close together), and fixed the issue where text like `a day ago` could be widowed (e.g. `a day \n ago`).

This should be ready for a quick re-review! Thanks, @Osmose !

fitojb commented:
The extremely thin font used in small copy (buttons, footer, etc.) is completely uncalled for. Do you really think this is readable?
Thin typographical weights are designed for display sizes, not body copy.
(Screenshot taken under Ubuntu)
![Unreadable thin font](https://user-images.githubusercontent.com/554953/39811335-f36adfaa-534d-11e8-9407-a2a0a7814e78.png)

fitojb commented:
This is [offensive](https://www.telegraph.co.uk/science/2016/10/23/internet-is-becoming-unreadable-because-of-a-trend-towards-light/). ↓
![captura de pantalla de 2018-05-09 06-00-21](https://user-images.githubusercontent.com/554953/39811434-4b916992-534e-11e8-8c26-a4792f0645a5.png)

willkg commented:
@fitojb Please open up a bug in the bug tracker: https://bugzilla.mozilla.org/enter_bug.cgi?format=__standard__&product=Socorro
